lib/cog/repository/bundles.ex

```diff
@@ -631,6 +631,40 @@ defmodule Cog.Repository.Bundles do
     |> preload
   end
 
+  defp update_version!(version, params) do
+    params = Map.merge(params, Map.take(params["config_file"], ["author", "homepage"]))
+
+    version
+    |> BundleVersion.changeset(params)
+    |> Repo.update!
+    |> preload
+  end
+
+  defp new_or_updated_version!(bundle, params) do
+    # Check for an old version
+    version = params["config_file"]["version"]
+    result =
+      Repo.one(from bv in BundleVersion,
+               where: bv.bundle_id == ^bundle.id,
+               where: bv.version == ^version)
+
+    case result do
+      nil ->
+        # Create a new version if there isn't one
+        new_version!(bundle, params)
+      old_version ->
+        # If there is a version update it and then delete any commands and
+        # templates associated with it.
+        new_version = update_version!(old_version, params)
+        |> Repo.preload(:templates)
+
+        Enum.each(new_version.commands, &Repo.delete!/1)
+        Enum.each(new_version.templates, &Repo.delete!/1)
```

Comment:
Hrm, that would be cleaner. I did it like this to preserve the id of the bundle version. That way we wouldn't have to go through and clean up references to it. But as I'm looking again you're right. The only one to worry about is the enabled version. Everything else references the bundle id.

Let me try that and see how if everything works like I'm thinking.